Gravity/Gravity.Test.Integration/RSAPI_IntegrationTest.cs

```diff
@@ -0,0 +1,395 @@
+
+                //Assert
+                Console.WriteLine("Starting Assertion....");
+
+                //Assert
+                Assert.AreEqual(sampleData, newObjectValue);
+
+                Console.WriteLine("Assertion Complete....");
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Error encountered in " + System.Reflection.MethodBase.GetCurrentMethod().Name, ex);
+            }
+            finally
+            {
+                Console.WriteLine("Ending Test case " + System.Reflection.MethodBase.GetCurrentMethod().Name);
+            }
+        }
+
+        [TestCase("LongTextField", TestValues.LongTextFieldValue)]
+        [TestCase("FixedTextField", TestValues.String100Length)]
+        [TestCase("IntegerField", -1)]
+        [TestCase("BoolField", true)]
+        [TestCase("DecimalField", 123.45)]
+        [TestCase("CurrencyField", 5648.54)]
+        public void Valid_Gravity_Object_Read_Field_Type<T>(string objectPropertyName, T sampleData)
+        {
+            Console.WriteLine(System.Reflection.MethodBase.GetCurrentMethod().Name + " Created");
+
+            try
+            {
+                //Arrange
+                Console.WriteLine("Starting Arrangement for property...." + objectPropertyName);
+
+                GravityLevelOne testObject = new GravityLevelOne();
+                testObject.Name = "TestObjectRead_" + objectPropertyName + Guid.NewGuid().ToString();
+
+                Guid testObjectTypeGuid = testObject.GetType().GetCustomAttribute<RelativityObjectAttribute>(false).ObjectTypeGuid;
+                Guid nameFieldGuid = BaseDto.GetCustomAttributeOfProperty<GravityLevelOne, Guid>("Name", x => x.FieldGuid);
+                Guid testFieldGuid = BaseDto.GetCustomAttributeOfProperty<GravityLevelOne, Guid>(objectPropertyName, x => x.FieldGuid);
+                RdoFieldType fieldType = (RdoFieldType)BaseDto.GetCustomAttributeOfProperty<GravityLevelOne, int>(objectPropertyName, x => x.FieldType);
+
+                _client.APIOptions.WorkspaceID = _workspaceId;
+
+                var dto = new RDO();
+                int newArtifactId = -1;
+                dto.ArtifactTypeGuids.Add(testObjectTypeGuid);
+                dto.Fields.Add(new FieldValue(nameFieldGuid, testObject.Name));
+                dto.Fields.Add(new FieldValue(testFieldGuid, sampleData));
+                WriteResultSet<RDO> writeResults = _client.Repositories.RDO.Create(dto);
+
+                if (writeResults.Success)
+                {
+                    newArtifactId = writeResults.Results[0].Artifact.ArtifactID;
+                    Console.WriteLine(string.Format("Object was created with Artifact ID {0}.", newArtifactId));
+                }
+                else
+                {
+                    Console.WriteLine(string.Format("An error occurred creating object: {0}", writeResults.Message));
+
+                    for (Int32 i = 0; i <= writeResults.Results.Count - 1; i++)
```

Comment:
Even if you need the indices, you should still not use a for loop. You can do something like the following:

`foreach (failure in writeResults.Results.Select((Result, Index) => new {Result, Index}}.Where(x => !x.Result.Success))`

Reply:
Also, I would modify `GetResultData` to return throw an AggregateException with one exception per failure, then catch and log that. That would reduce the code here and give that detailed logging info to all consumers.

Reply:
We can do this later though, if that's easier.

Reply:
It would be easier, but not necessarily right.  I appreciate you holding me accountable on this stuff.  I got it and committed the changes.  Definitely making the product (and me) better!